### PR TITLE
Update Mill to 0.10.13, cleanup and fix deprecations

### PR DESCRIPTION
**Pull request by lefou**
- Update Mill to 0.10.13
- Some obvious review fixes and updates
- Enable some warnings and deprecations in build scripts
- Fixed various compiler warnings
- Replaced deprecated use of `Tests` traits

### CODE REVIEW

File: project/modules/cache0.sc [hunk 1]
```diff
@@ -59,4 +59,6 @@ trait CacheJvmBase extends Cache with CsMima with Shading {
       super.sources() ++ Seq(PathRef(dest))
     }
   }
+  @deprecated("Use CacheJvmBaseTests")
+  trait Tests extends CacheJvmBaseTests
```
Comment (tgodzik) [inline project/modules/cache0.sc:63]: Could we remove it altogether? It's only used in the coursier build, no?

**Review by tgodzik — COMMENTED** [review]
Looks good! Just one question.

Do you want to add anything more here?

File: project/modules/cache0.sc (hunk 1 above)
Comment (lefou) [reply project/modules/cache0.sc:63]: Yeah, I needed it in-between to let the compiler give me more precise messages. I'll remove later.

**Comment by lefou** [comment]
> Looks good! Just one question.
> 
> Do you want to add anything more here?

I want to migrate to Mill 0.11, but this is still a bit far, so a merge now makes sense. I'll remove the draft state once I think I'm done.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**